### DISCUSSION TITLE
Bluetooth service error on boot

### DISCUSSION THREAD

**Discussion by HafizBeyy** [blueman-project/blueman#2288] [Q&A]
Mint 21.3
Kernel: 5.15.0-97-generic
every time i boot up my pc , bluetooth interface refuses to work. i restart and then it works flawlesly.
[    6.142254] Bluetooth: Core ver 2.22
[    6.142572] Bluetooth: HCI device and connection manager initialized
[    6.142578] Bluetooth: HCI socket layer initialized
[    6.142580] Bluetooth: L2CAP socket layer initialized
[    6.142586] Bluetooth: SCO socket layer initialized
[    6.163918] Bluetooth: hci0: CSR: Setting up dongle with HCI ver=9 rev=3120; LMP ver=9 subver=22bb; manufacturer=10
[    6.163922] Bluetooth: hci0: CSR: Unbranded CSR clone detected; adding workarounds and force-suspending once...
[   11.402522] Bluetooth: hci0: CSR: Failed to suspend the device for our Barrot 8041a02 receive-issue workaround
[   13.534435] Bluetooth: BNEP (Ethernet Emulation) ver 1.3
[   13.534439] Bluetooth: BNEP filters: protocol multicast
[   13.534442] Bluetooth: BNEP socket layer initialized
**[   16.520428] Bluetooth: hci0: setting interface failed (110)**

**Comment by jeremyb31** [comment]
This is a kernel issue, it is not caused by Blueman

**Comment by cschramm** [comment]
Yes, your hardware and kernel do not like each other. I'd ether try a recent kernel or find our more about the hardware. There are plenty of examples for fake CSR chips with those symptoms on the Internets.

**Reply by HafizBeyy** [reply]
which kernel do you recommend?

**Reply by jeremyb31** [reply]
It is hard to say with those devices, CSR has been a problem in Linux for years, I use an Edimax BT8500 BT dongle without issues

**Reply by cschramm** [reply]
https://lore.kernel.org/linux-bluetooth/20220307200445.5554-1-swyterzone@gmail.com/ is one thing that could match. It should be part of 5.18 and later. https://gist.github.com/nevack/6b36b82d715dc025163d9e9124840a07 is a decent source for similar cases.

The underlying problem is that there is an endless number of fake CSR chips that Linux handles like what they identify themselves as but they simply are not and have an endless number of quirks. I rather wonder why they wouldn't be a problem to other systems (the simple answer might be that they all come with their own Windows driver that you have to install :facepalm:).

**Reply by HafizBeyy** [reply]
ok im gonna try

**Reply by jeremyb31** [reply]
Use Mints Update Manager, view menu, linux kernels, select 6.5 on the left and install the newest. Then reboot

**Reply by HafizBeyy** [reply]
> Use Mints Update Manager, view menu, linux kernels, select 6.5 on the left and install the newest. Then reboot 

~~Thanks!~~

**Reply by HafizBeyy** [reply]
https://media.discordapp.net/attachments/1034529384327020647/1216063686918017137/InShot_20240309_194154002.mp4?ex=65ff0690&is=65ec9190&hm=2a60039812233c911a26aec9849d18d4ff546fb4058deda67acd3535a6eb0620&

**Reply by infirit** [reply]
Assuming this is a usb dongle, where did you buy it and how much was it? Usually these fake chips devices are $5 or less and they never work well.

**Reply by HafizBeyy** [reply]
im from turkey and i bought it a few months ago, in a local online website "trendyol". it was 40 turkish liras, so its highly possible to being one of fake ones. but it **was** working properly when i reboot the system on kernel 5.15.97, but in this kernel (6.15), i dont get a error like the previous one but still it refuses to work.

**Reply by HafizBeyy** [reply]
i just wanted to dont reboot system everytime for my dongle to work.

**Reply by infirit** [reply]
I understand the frustration but this is most likely how things will be with these dongles. A dongle with a proper CSR chip (at least last time I checked) is this one https://www.trendyol.com/tp-link/ub400-bluetooth-4-0-mini-usb-adaptor-p-6814005 which is more than 5 times the price you paid (40 tl is about 1.30 usd).

**Reply by cschramm** [reply]
https://bugzilla.kernel.org/show_bug.cgi?id=60824 does not sound promising. Depending on the precise chip, it might work starting with Linux 5.17, but stopped working with 6.0 again.
